File: backend/src/main/java/com/campus/club/entity/Checkin.java
package com.campus.club.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("checkins")
public class Checkin {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long activityId;
    private Long userId;
    private LocalDateTime checkinTime;
    private String remark;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/checkin.js
import request from '../utils/request'

export function checkin(activityId, userId, remark) {
  return request({
    url: `/activities/${activityId}/checkins`,
    method: 'post',
    data: { userId, remark }
  })
}

export function getActivityCheckins(id) {
  return request({
    url: `/activities/${id}/checkins`,
    method: 'get'
  })
}

export function getCheckinStatus(id) {
  return request({
    url: `/activities/${id}/checkins/status`,
    method: 'get'
  })
}
